include/deal.II/base/exceptions.h

```diff
@@ -28,6 +28,9 @@
 #  include <cusparse.h>
 #endif
 
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#include <Kokkos_Core.hpp>
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
```

Comment:
This file, when preprocessed, is 100k lines - that's a lot to add to every single file which includes `exceptions.h`. Can we get away with doing something more minimal?

Reply:
I didn't try it but maybe `Kokkos_Macros.hpp` is enough.

Reply:
We need `Kokkos_Core.hpp` for `Kokkos::abort`. We could define another (device-callable) function taking a `const char *` to wrap `Kokkos::abort` that we use and put its definition in a source file. We certainly don't care about it being inlined anyway.